src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -487,13 +527,25 @@ ssize_t uct_dc_mlx5_ep_put_bcopy(uct_ep_h tl_ep, uct_pack_callback_t pack_cb,
 
     UCT_DC_MLX5_CHECK_RES(iface, ep);
     UCT_RC_IFACE_GET_TX_PUT_BCOPY_DESC(&iface->super.super, &iface->super.super.tx.mp,
-                                       desc, pack_cb, arg, length);
+                                       desc, pack_cb, arg, length,
+                                       {
+                                           uct_dc_mlx5_iface_dci_put(iface,
+                                                                     ep->dci);
+                                           return UCS_ERR_NO_RESOURCE;
+                                       });
+    UCT_SKIP_ZERO_LENGTH_ACTION(length,
+                                uct_dc_mlx5_iface_dci_put(iface, ep->dci),
+                                desc);
     UCT_DC_MLX5_IFACE_TXQP_GET(iface, ep, txqp, txwq);
+
     uct_rc_mlx5_ep_fence_put(&iface->super, txwq, &rkey, &remote_addr,
                              ep->atomic_mr_offset);
     uct_dc_mlx5_iface_bcopy_post(iface, ep, MLX5_OPCODE_RDMA_WRITE, length,
                                  remote_addr, rkey, desc, 0, 0, desc + 1, NULL);
+
     UCT_TL_EP_STAT_OP(&ep->super, PUT, BCOPY, length);
+    uct_dc_mlx5_ep_check_dci_leak(ep);
```

Comment:
what is the point to check for leaks in sending functions? You will always have something outstanding in the end of function.

Reply:
removed